Multiple issues

jordins opened the pull request:
## Notify on filepath change

We will attach this macro to Open Office "Title changed" event.
Each time we open a new document or we save it with another name it will get the new name and send it to the desktop.

We want it because we want the desktop to display always the correct file in the url. Doing so when the user refreshes the page will always see the correct file.

vHanda committed:
Connect to LO via a pipe instead of a tcp socket

We do not need the extra overhead of using a TCP socket. For a more
detailed reasoning see 28876ada in open365-base-with-dependencies

Commit:
fix /car/lib => /var/lib

Commit:
Refactor/Pythonify LibreOffice scripts

Extract common components of autosave/closeApps scripts
Fix indentation (to 4 spaces)
Remove semicolons
add if __name__ == "__main__" guards

Commit:
Add stand-alone script to save all documents

Commit:
Fix file permissions

Executable bit was missing

jordins committed:
Create open office macro that calls script to notify filepath changes

We will attach this macro to Open Office "Title changed" event.
Each time we open a new document or we save it with another name it will
get the new name and send it to the desktop.

We want it because we want the desktop to display always the correct
file in the url. Doing so when the user refreshes the page will always
see the correct file.

afiestas committed:
Install our own libreoffice, located in amazon

Ideally we should have our own debian repository, that will eventually
happened.

Atm we have a tar with all our packages for the libreoffice built with
customizations, not ideal but it works.

afiestas committed:
Install only needed packages instead of all libreoffice things

Otherwise the image is 6Gb large and we do not need everything anyway.

jordins committed:
Remove new line

jordins committed:
Avoid crashes in office macro that notifies document saved

This error happened each time you opened a new document, because
hasLocation method of ThisComponent didn't exist yet.

We just ignore all the errors.

vHanda commented on scripts/LOffice.py:
```diff
@@ -0,0 +1,57 @@
+            logger.warning("Could not connect to LibreOffice")
+            time.sleep(1)
+
+def saveAllDocs(desktop):
```
This function can crash.

The connection to LO could be invalidated for a small amount of time and then when we try and use this connection, both LibreOffice and this script can crash. It's hard to reproduce, since it requires things to line up perfectly.

Could you please refactor the code so that these functions always first connect, then perform the action and then disconnect. Making a connection should be cheap given that we're using a pipe.

jordins committed:
Modify libreoffice skel files to add macro that notifies file path changes

We attach this macro to Open Office "Title changed" event. Each time we
open a new document or we save it with another name it will get the new
name and send it to the desktop.

We want it because we want the desktop to display always the correct
file in the url. Doing so when the user refreshes the page will always
see the correct file.

jordins committed:
Create migration system

Each time the container is run we execute the migrations from the folder
/usr/lib/open365/migrations.d

jordins committed:
Create libreoffice migration "notifyFilePathChanged"

Copies the modified libreoffice config files to the users that do not
have them.

jordins committed:
Fix libreoffice config copy to correct place

Without it, new users are never going to have our default libreoffice
config files.

jordins committed:
Add libreoffice default config for 5.0.5.2

Since we've downgraded libreoffice version we need the default config
files to be regenerated.

jordins committed:
Apply macro again (in downgraded version) to default user config

Since we've downgraded to a previous libreoffice version and regenerated
the config files, we need to apply again our custom modifications.

These modifications right now are only a macro executed on "document
title change". It send a message through the bus with the new file path.

jordins committed:
Copy the whole skel folder the first time

Because we've moved the "migration" that was in base-with-dependencies
repo to here. That migration was deleting libreoffice config because we
downgraded libreoffice version.

jordins committed:
Disabled ctrl+click to follow hiperlinks

We just want click to follow hiperlinks.

jordins committed:
Improve condition to run migration

Instead of doing a grep we create a file that marks if the migration has
been done.
This is done because checking if a file exist is faster than doing a
grep.

jordins committed:
Fix libreoffice error message after doing 2 consecutive "save as"

Due to a bug in libreoffice when a macro is attached to "document title
change" event, and doing "save as" more than once, it shows an error
message:
"com.sun.star.uno.RuntimeExceptionScriptProtocolHandler:createScriptProvider()"

Since we can't use that event, we use "SavedAs" and "ActiveDocument" to
accomplish the same result.

Commit:
add .dockerignore

Using this our images will be thinner because they won't add those
files/folders.

Commit:
Create a new connection to libreOffice each time

This avoids some race conditions that can make libreoffice crash

ferranmesas commented:
I chaned the autosave.py script (the only one using long-running connections to libreOffice). It now creates a new connection for each autosave attempt.
